docs/src/guide/capabilities.md

```diff
@@ -86,16 +86,16 @@ That's it for linking the capability into our app, now we can use it in the `upd
     // ...
 ```
 
-You can see the use of the `Event::Set` variant we just discussed. `Event::Set` is techincally a function with this signature:
+You can see the use of the `Event::Set` variant we just discussed. `Event::Set` is technically a function with this signature:
 
 ```rust
 fn Event::Set(crux_http::Result<crux_http::Response<Counter>) -> Event
 ```
 
 Looks a lot like a callback, doesn't it. Yep. With the difference that the result is an `Event`. Generally, you should be able to completely ignore this detail and just use your variant names and the code should read pretty clearly: "When done, send me `Event::Set`".
 
-The other nuance to be aware of is that the capability calls return immediately. This should hopefully be relatively obvious by now, but all that's happening is effects are getting queued up to be requested from the Shell. In a way, capability calls are implicitly asynchronous (but you can't await them). It's perhaps easier to think about the capability calls as a kind of `return` statements which you can call within your update function and add effects to the batch to request.
+The other nuance to be aware of is that the capability calls return immediately. This should hopefully be relatively obvious by now, but all that's happening is effects are getting queued up to be requested from the Shell. In a way, capability calls are implicitly asynchronous (but you can't await them). It's perhaps easier to think about a capability call as a kind of `return` statement (or even better, a `yield` statement), which you can call within your update function, and which adds an effect to the batch that is sent to the shell for execution.
```

Comment:
I thought about mentioning a `yield`, but both analogies have the same kind of problem that they also imply control flow, they either interrupt the code, or terminate it ealry, neither of which happens with the capability calls. Wonder if the whole comparison confuses matters, rather than clarify them...

Reply:
Yeah good point. I think the comparison is valid, but probably needs that clarification.

Reply:
or maybe, you're right and save it for when talking about building the capability itself